internal/attestation/idkeydigest/idkeydigest.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package idkeydigest
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
+	"go.uber.org/multierr"
+)
+
+// IDKeyDigests is a list of trusted digest values for the ID key.
+type IDKeyDigests [][]byte
+
+type encodedIDKeyDigests []string
+
+// NewIDKeyDigests creates a new IDKeyDigests from a list of digests.
+func NewIDKeyDigests(digests [][]byte) IDKeyDigests {
+	idKeyDigests := make(IDKeyDigests, len(digests))
+	copy(idKeyDigests, digests)
+	return idKeyDigests
+}
+
+// DefaultsFor returns the default IDKeyDigests for the given cloud provider.
+func DefaultsFor(csp cloudprovider.Provider) IDKeyDigests {
+	switch csp {
+	case cloudprovider.Azure:
+		return IDKeyDigests{
+			{0x57, 0x48, 0x6a, 0x44, 0x7e, 0xc0, 0xf1, 0x95, 0x80, 0x02, 0xa2, 0x2a, 0x06, 0xb7, 0x67, 0x3b, 0x9f, 0xd2, 0x7d, 0x11, 0xe1, 0xc6, 0x52, 0x74, 0x98, 0x05, 0x60, 0x54, 0xc5, 0xfa, 0x92, 0xd2, 0x3c, 0x50, 0xf9, 0xde, 0x44, 0x07, 0x27, 0x60, 0xfe, 0x2b, 0x6f, 0xb8, 0x97, 0x40, 0xb6, 0x96},
+			{0x03, 0x56, 0x21, 0x58, 0x82, 0xa8, 0x25, 0x27, 0x9a, 0x85, 0xb3, 0x00, 0xb0, 0xb7, 0x42, 0x93, 0x1d, 0x11, 0x3b, 0xf7, 0xe3, 0x2d, 0xde, 0x2e, 0x50, 0xff, 0xde, 0x7e, 0xc7, 0x43, 0xca, 0x49, 0x1e, 0xcd, 0xd7, 0xf3, 0x36, 0xdc, 0x28, 0xa6, 0xe0, 0xb2, 0xbb, 0x57, 0xaf, 0x7a, 0x44, 0xa3},
+		}
```

Comment:
Could we keep the values as strings and return the result of `hex.EncodeToString(..)`? That would be much easier to update imo.

Reply:
See [my earlier comment](https://github.com/edgelesssys/constellation/pull/991#discussion_r1072377533) on the issue.
The underlying code works on `[]byte` anyways, so I don't see any benefit from keeping it as hex encoded string until we get to the point where it is actually used.

If our checking tool generates the output as we expect it in the source code, there should be no difference in ease of updates.